platform/src/test/kotlin/org/stellar/anchor/platform/configurator/ConfigManagerTest.kt

```diff
@@ -30,17 +30,17 @@ class ConfigManagerTest {
 
     val config = configManager.processConfigurations(null)
     assertEquals(config.get("languages").value, "tw, en, fr")
-    assertEquals(config.get("clients.noncustodial[0].name").value, "vibrant")
-    assertEquals(config.get("clients.noncustodial[0].domains[0]").value, "vibrant.co")
+    assertEquals(config.get("clients.items[0].name").value, "vibrant")
+    assertEquals(config.get("clients.items[0].domains[0]").value, "vibrant.co")
     assertEquals(
-      config.get("clients.noncustodial[0].callback_urls.sep24").value,
+      config.get("clients.items[0].callback_urls.sep24").value,
       "https://callback.vibrant.com/api/v2/anchor/callback/sep24"
     )
 
-    assertEquals(config.get("clients.noncustodial[1].name").value, "lobstr")
-    assertEquals(config.get("clients.noncustodial[1].domains[0]").value, "lobstr.co")
+    assertEquals(config.get("clients.items[1].name").value, "lobstr")
+    assertEquals(config.get("clients.items[1].domains[0]").value, "lobstr.co")
     assertEquals(
-      config.get("clients.noncustodial[1].callback_urls.sep6").value,
+      config.get("clients.items[1].callback_urls.sep6").value,
```

Comment:
Do we still support setting client configs with environment variables?

Reply:
My understanding is that users can always set these through environment variables as long as they use the correct format. This is because all file/inline configurations are ultimately read into the environment, and in client config case `clients.noncustodial[1].callback_urls.sep6` is how it is preserved in the environment variables.

Reply:
I had trouble setting the callback URLs when it is nested under `callback_urls`. That's why in 2.9.0, I opted to make this configuration flat fields `callback_url_sep_*`. Can you test if it can be overridden using the `.env` file?

Reply:
Ok I know where went wrong...

When load config from env var, [`loadConfigFromEnv()`](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/main/java/org/stellar/anchor/platform/configurator/ConfigHelper.java#L45), specifically in [`ConfigEnvironment`](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/main/java/org/stellar/anchor/platform/configurator/ConfigHelper.java#L76), changed the var name from `clients.items[0].abc` to `clients_items[0].abc`, and this makes it failed validation against config schema.
transformation is like `clients.items[0].abc` ->

Reply:
We can implement a workaround specifically for the callback URLs if a general solution is too difficult.

Although I'm ok with delaying inline support.

Reply:
We have the plan to migrate from Spring `@ConfigurationProperty` to straight YAML parsing. [ANCHOR-803](https://stellarorg.atlassian.net/browse/ANCHOR-803)